src/components/POLine/POLineDetails/FieldReceiptStatus.js

```diff
@@ -13,28 +14,53 @@ export const RECEIPT_STATUS = {
   receiptNotRequired: 'Receipt Not Required',
 };
 
+const RECEIPT_STATUSES_BY_ORDER_STATUS = {
+  Pending: [
+    'pending',
+    'receiptNotRequired',
+  ],
+  Open: [
+    'partiallyReceived',
+    'receiptNotRequired',
+    'fullyReceived',
+    'cancelled',
+  ],
+};
+
 class FieldReceiptStatus extends Component {
   render() {
+    const { workflowStatus } = this.props;
+
     return (
       <Field
         component={Select}
         label={<FormattedMessage id="ui-orders.poLine.receiptStatus" />}
         name="receiptStatus"
+        disabled={!RECEIPT_STATUSES_BY_ORDER_STATUS[workflowStatus]}
       >
-        <FormattedMessage id="ui-orders.dropdown.select">
-          {(message) => <option value="">{message}</option>}
-        </FormattedMessage>
-        {Object.keys(RECEIPT_STATUS).map((key) => (
-          <FormattedMessage
-            id={`ui-orders.receipt_status.${key}`}
-            key={key}
-          >
-            {(message) => <option value={RECEIPT_STATUS[key]}>{message}</option>}
-          </FormattedMessage>
-        ))}
+        {RECEIPT_STATUSES_BY_ORDER_STATUS[workflowStatus]
+          ? (
+            <React.Fragment>
+              <option value="" />
+              {RECEIPT_STATUSES_BY_ORDER_STATUS[workflowStatus].map((key) => (
+                <FormattedMessage
+                  id={`ui-orders.receipt_status.${key}`}
+                  key={key}
+                >
+                  {(message) => <option value={RECEIPT_STATUS[key]}>{message}</option>}
+                </FormattedMessage>
+              ))}
+            </React.Fragment>
+          )
+          : (<option value="" />)
```

Comment:
you have this empty option in either case, maybe it makes sense to put it above the condition?

Reply:
I tried to do like that that, but it doesn't work properly. Selector component expected only <option> child.

Reply:
thank you for explanation